autogpt_platform/backend/backend/data/block.py

```diff
@@ -492,6 +510,127 @@ def get_blocks() -> dict[str, Type[Block]]:
     return load_all_blocks()
 
 
+def is_block_provider_configured(block_cls: type["Block"]) -> bool:
+    """
+    Check if a block has a valid authentication method configured at runtime.
+
+    For example if a block is an OAuth-only block and there env vars are not set,
+    do not show it in the UI.
+
+    """
+    from backend.sdk.registry import AutoRegistry
+
+    # Create an instance to access input_schema
+    try:
+        block = block_cls.create()
+    except Exception as e:
+        # If we can't create a block instance, assume it's not OAuth-only
+        logger.error(f"Error creating block instance for {block_cls.__name__}: {e}")
+        return True
+    logger.debug(
+        f"Checking if block {block_cls.__name__} has a valid provider configured"
+    )
+
+    # Get all credentials fields from input schema
+    credentials_fields = block.input_schema.get_credentials_fields()
+    if not credentials_fields:
+        logger.debug(
+            f"Block {block_cls.__name__} has no credentials fields - Treating as valid"
+        )
+        return True
+
+    # Check required credentials fields
+    required_fields = block.input_schema.get_required_fields()
+    required_credentials = [f for f in credentials_fields if f in required_fields]
+
+    if not required_credentials:
+        logger.debug(
+            f"Block {block_cls.__name__} has no required credentials fields - Treating as valid"
+        )
+        return True
+
+    # Check if ANY required credentials field has authentication configured
+    configured_providers = []
+
+    for field_name in required_credentials:
+        provider_name = block.input_schema.get_provider_from_credentials_field(
+            field_name
+        )
+        if not provider_name or provider_name in ProviderName.__members__.values():
+            logger.debug(
+                f"Block {block_cls.__name__} is part of the legacy provider system - Treating as valid"
+            )
+            return True
+
+        provider = AutoRegistry.get_provider(provider_name)
+        if not provider:
+            logger.debug(
+                f"Block {block_cls.__name__} has a required credentials field {field_name} that has no provider - Treating as valid"
+            )
+            continue
+
+        # Check the provider's supported auth types
+        supported_types = provider.supported_auth_types
+
+        # If provider has no supported auth types, it means no auth method is configured
+        if not supported_types:
+            # This provider has no authentication methods configured at all
+            configured_providers.append(False)
```

Comment:
If any required credentials field doesn't have any supported auth types configured, the block is unusable, so we can go straight to:
```suggestion
            return False
```